hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/command/SqlKeyGenerator.scala

```diff
@@ -53,8 +55,17 @@ class SqlKeyGenerator(props: TypedProperties) extends ComplexKeyGenerator(props)
       keyGenProps.remove(SqlKeyGenerator.ORIGIN_KEYGEN_CLASS_NAME)
       val convertedKeyGenClassName = SqlKeyGenerator.getRealKeyGenClassName(props)
       keyGenProps.put(HoodieWriteConfig.KEYGENERATOR_CLASS_NAME.key, convertedKeyGenClassName)
```

Comment:
SqlKeyGenerator as the default keyGenerator for Spark SQL. The change below checks if `encodePartition` config is turned on.